docs/wip/build-pipeline.md

```diff
@@ -1,7 +1,95 @@
+
+![Store Framework Jamstack CI checks](./../images/queued_check_runs.png)
+
+
+## Build
+
+This step is responsible for two main things:
+
+### Build the store
+- Run an automated build for the store, saving all the generated artifacts in our infrastructure.
+- Send the build logs to GitHub.
+The build logs are visible by clicking on `details` on the `Build` check run once it's completed.
+
+![Store Framework Jamstack CI build check completed](./../images/build_check_completed.png)
+
+
+#### Possible states
+- `queued`: soon this step will be executed.
+- `in_progress`: this step is being executed.
+- `completed/success`: the build is completed successfully.
+- `completed/failure`: the build is completed but something went wrong.
+
+### Generate a deploy preview 
+For each pull request, we provide a `Deploy Preview` that is a deployment of the most recent changes of the pull request.
+You can access the `Deploy Preview` of a specific Pull Request by clicking on `details` on the commit status with the title `<your store>/deploy-preview`:  
+
+![Store Framework Jamstack CI deploy preview commit status](./../images/deploy_preview_commit_status.png)
+
+Or directly on this url: `https://preview-<pull request number>--<your store name>.vtex.app/`.  
+For example: `https://preview-417--storecomponents.vtex.app/`.
+
+
+## SonarQube
+
+[SonarQube](sonarqube.md) is used to measure and analyze code quality using static analysis.
+
+Every time that the CI flow is triggered, a static analysis of the code is made taking into account the latest changes of the pull request. If any issues are found by sonarqube, they will be marked as annotations on the pull request like so:
+
+![Store Framework Jamstack CI sonarqube annotations](./../images/sonarqube_annotation.png)
+
+#### Possible states
+- `queued`: soon this step will be executed.
+- `in_progress`: this step is being executed.
+- `completed/success`: Sonarqube didn't found any critical issues.
+- `completed/failure`: There is at least one critical issue found by sonarqube.
+
+More information can be found [here](./sonarqube.md)
 
 ## Lighthouse CI
 
-## Explain each CI that we do
+Lighthouse is a tool that is used to monitor the performance of a website, and it is executed each time a new build is completed, using the `Deploy Preview` as the source of analysis. These results are aggregated and allow the developer to compare the performance of whats being developed with what is in production.  
+
+You can access the execution logs by clicking on `details` at the `Lighthouse` check.  
+![Store Framework Jamstack CI lhci check completed](./../images/lhci_check_completed.png)
+
+#### Possible states
+- `queued`: soon this step will be executed.
+- `in_progress`: this step is being executed.
+- `completed/success`: Lighthouse CI is completed successfully.
+- `completed/failure`: Something went wrong.
+- `completed/canceled`: The build was not successful, therefore the Lighthouse CI can't be executed.
+
```

Comment:
```suggestion
```

Reply:
It is important to explicit the possible states of LHCI and Integration tests steps as they are influenced by the result of the Build Step.